homeassistant/components/matter/api.py

```diff
@@ -173,3 +180,124 @@ async def websocket_set_wifi_credentials(
         ssid=msg["network_name"], credentials=msg["password"]
     )
     connection.send_result(msg[ID])
+
+
+@websocket_api.websocket_command(
+    {
+        vol.Required(TYPE): "matter/node_diagnostics",
+        vol.Required(DEVICE_ID): str,
+    }
+)
+@websocket_api.async_response
+@async_handle_failed_command
+@async_get_matter_adapter
+async def websocket_node_diagnostics(
+    hass: HomeAssistant,
+    connection: ActiveConnection,
+    msg: dict[str, Any],
+    matter: MatterAdapter,
+) -> None:
+    """Gather diagnostics for the given node."""
+    node = node_from_ha_device_id(hass, msg[DEVICE_ID])
+    if not node:
+        raise MatterError("Node not found")
```

Comment:
Maybe include which node wasn't found?

We also shouldn't raise an exception from the library here, as those are not meant to be raised outside of the library.

Reply:
suggestion for an exception to raise ?

Reply:
```py
class MissingNode(HomeAssistantError):
```

Reply:
Yeah, I was just checking and a bit below i the helper we actually raise ValueError if the device can not be found so maybe raise ValueError here as well to be consistent ?

Reply:
If we want to catch the exception I'd raise a more specific error. I assume we want to catch the exception in the error handler decorator and send an error message to the frontend.

The raised errors in the helpers are not expected but more developer bugs if they would happen.

Reply:
can you check what I did - I'm now raising ValueError with a specific error message and catch that in the decorator

Reply:
yeah sorry, forgot to hit push. its now there